wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/wsplugins/model/ExtendedPluginFQN.java

```diff
@@ -34,11 +34,13 @@ public ExtendedPluginFQN(URI registry, String id, String publisher, String name,
     this.version = version;
   }
 
+  /** In this constructor, {@link PluginFQN#id} is composed from given params */
   public ExtendedPluginFQN(String reference, String publisher, String name, String version) {
     super(reference);
     this.publisher = publisher;
     this.name = name;
     this.version = version;
+    this.id = publisher + "/" + name + "/" + version;
```

Comment:
Setting `id` when a `reference` is set directly contradicts the javadoc on `PluginFQN`:

```java
/**
 * Represents full information about plugin, including registry address and id, or direct reference
 * to plugin descriptor. Should NOT contain both reference and id simultaneously.
 *
```
This should be updated, and add a note that if both are defined, reference takes precedence.

I've checked the places I'm aware of `PluginFQN` being used and I don't think this will cause issues.

Reply:
done. I've also created new protected constructor, instead of setting id field protected.